wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -621,6 +621,16 @@ def required_formsets(cls):
             }
         }
 
+    @classmethod
+    def html_declarations(cls):
+        # We need to include html_declarations for StreamFields in the related model
+        declarations = [
+            field.stream_block.all_html_declarations()
+            for field in cls.related.related_model._meta.fields
+            if hasattr(field, 'stream_block')
+        ]
```

Comment:
We really shouldn't have any special-case code for StreamField (or any other field type) here - that would imply that something in the StreamField mechanism is failing to follow the same spec as everything else. Investigating...